### PR TITLE
refactor: improved the nginx shut down logic. thanks Dejiang Zhu for the suggestion

### PR DESCRIPTION
**Pull request by agentzh**

### CODE REVIEW

**Comment by agentzh** [comment]
@doujiang24 @membphis Please have a look. Thanks!

File: lib/Test/Nginx/Util.pm [hunk 1]
```diff
@@ -1430,25 +1438,7 @@ sub run_test ($) {
...
-                            last unless is_running($pid);
-
-                            sleep $TestNginxSleep;
-                            next if $i < $max_i;
-
-                            warn "WARNING: $name - killing nginx $pid with force...";
-                            kill(SIGKILL, $pid);
-                            waitpid($pid, 0);
-                        }
+                        kill_process($pid, 1, $name);
```
Comment (doujiang24) [inline lib/Test/Nginx/Util.pm]: we should `kill_process($pid, 0, $name);` to kill nginx?
For now, it won't kill force.

Comment (agentzh) [inline lib/Test/Nginx/Util.pm]: @doujiang24 Just committed a fix to this branch to ensure `kill_process` always kills the process with force as the last resort. Please check it out. Thanks!

Comment (membphis) [inline lib/Test/Nginx/Util.pm]: the second parameter of `kill_process ` is always `1` now. Do we still need this parameter?

Comment (doujiang24) [inline lib/Test/Nginx/Util.pm]: @agentzh Works fine on my side now :)
@membphis I also think the second parameter is unnecessary now.

Comment (agentzh) [inline lib/Test/Nginx/Util.pm]: Right, we can retire the `wait` parameter in a future PR :)

**Comment by agentzh** [comment]
Already merged. Thanks for the review!

**Pull request merged**